rfcs/0137-nix-language-version.md

```diff
@@ -0,0 +1,117 @@
+---
+feature: nix-language-version
+start-date: 2022-12-12
+author: @fricklerhandwerk
+co-authors: @thufschmitt @Ericson2314
+shepherd-team: 
+shepherd-leader: 
+related-issues: https://github.com/NixOS/nix/issues/7255
+---
+
+# Summary
+[summary]: #summary
+
+Introduce a convention to determine which version of the Nix language grammar to use for parsing a Nix file.
+
+# Motivation
+[motivation]: #motivation
+
+Currently it's impossible to introduce backwards-incompatible changes to the Nix language without breaking existing setups.
+This proposal is first step towards overcoming that limitation.
```

Comment:
Limitations can be good and I would argue that the Nix language's stability requirements is a good limitation. The RFC needs to provide a better motivation than “allow backwards-compatible changes” – that backwards-incompatible changes are great and good by themselves is far from obvious.

Specifically the stability of Nix's language has been praised on multiple occasions, e.g. [Nix and legacy enterprise software development: an unlikely match made in heaven](https://www.youtube.com/watch?v=0dVxtNlD5N4) – evaluating a historical nixpkgs may continue to be possible.